src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,104 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.InvalidUserException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "user_login";
+    private static final String VALID_PASSWORD = "user_password";
+    private static final int MINIMUM_LOGIN_LENGTH = 4;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
+    private static final String INVALID_LOGIN = "log";
+    private static final String INVALID_PASSWORD = "1234";
+    private static final int INVALID_AGE = 17;
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User(VALID_LOGIN, VALID_PASSWORD, MINIMAL_AGE);
+    }
+
+    @BeforeEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_validUser_successfullyRegistered() {
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+        assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_nullUser_throwsIllegalArgumentException() {
+        user = null;
+        InvalidUserException actualException = assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user));
+        assertEquals("User cannot be null",actualException.getMessage());
+    }
+
+    @Test
+    void register_existingLogin_throwsInvalidLoginException() {
+        registrationService.register(user);
```

Comment:
We should not have used the register method to put the first user in the storage, because this method is under a test at this actual moment and can not be used to arrange the test data, because it is not guaranteed to work properly as expected, better to use low-level approaches like putting him there directly via `storageDao` or `Storage`
This is the point from common mistakes, check it for more details